### PR TITLE
Section4.php

### PR DESCRIPTION
**Pull request by rdripley**
@paulstatezny I made a pull request for section4.php I didn't see any questions but I changed "public" to "function" and the code ran the message that we wanted. Once you review the previous pull request, I figure we can work on the class thing.

### CODE REVIEW

File: php-practice/section4.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ class VideoGame {
   public $isMultiplayer; // Whether the game is a multiplayer game -- should be TRUE or FALSE
 
   // This is the contructor. Remember, this is the function that gets run whenever you create a new VideoGame object
-  public __construct($title, $publisher, $isMultiplayer) {
+  function __construct($title, $publisher, $isMultiplayer) {
```
Comment (paulstatezny) [inline php-practice/section4.php]: Good call on this -- it should have been `public function __construct`. (Although, if you don't specify the **visibility** [public/private] then it defaults to public for functions -- so omitting `public` works just as well.)

**Comment by paulstatezny** [comment]
@DrDoomsday Make sure to `git pull` before working on this because there were merge conflicts.

Bug me if it gives you trouble.

**Comment by paulstatezny** [comment]
Also, I finished section 4 as well as sections 5 through 9 -- we start getting into some fun stuff in section 5. :-) But let me know if I'm getting too advanced too quickly.

**Comment by rdripley** [comment]
@paulstatezny 
For question 2, I wasn't sure if you meant the "new" keyword since that is how you create a new object or the "__construct" keyword. Or if the keyword was "class".

For question 3, I'm assuming by method(function) you are referring to the "__construct" method.

File: php-practice/section4.php [hunk 2]
```diff
@@ -85,35 +84,36 @@ public function playCampaign() {
...
+ * ANSWER: I pass the arguments in the () next to the __construct keyword
  */
 
 /**
  * QUESTION 5
  *
  * How do you access a public property of an object?
- * ANSWER:
+ * ANSWER: We acess the public property of an object by echoing out the name of the varible (in this case $starCraft2) followed by -> and then the property we want to access 
+ * which would look like this - echo 'My favorite game is' . $starCraft2->title;
```
Comment (paulstatezny) [inline php-practice/section4.php:116]: Good answer.

By "access", I simply meant getting the value. Echoing out the value is an example situation where you would need to get the value. But you could do other things with the value, like...
1. Put it in a database
2. Use it to figure out further logic of your program.

etc...

File: php-practice/section4.php [hunk 3]
```diff
@@ -85,35 +84,36 @@ public function playCampaign() {
  * QUESTION 1
  *
  * What are the 2 main things that you can create inside of a class?
- * ANSWER:
+ * ANSWER: property variables and the constructor
```
Comment (paulstatezny) [inline php-practice/section4.php:87]: Good. I was actually looking for:
1. Properties (i.e. "class variables")
2. Methods (i.e. "class functions")

The constructor fits under 2.

But my wording of the question was pretty vague. :-P

**Comment by paulstatezny** [comment]
> For question 2, I wasn't sure if you meant the "new" keyword since that is how you create a new object or the "__construct" keyword. Or if the keyword was "class".
> 
> For question 3, I'm assuming by method(function) you are referring to the "__construct" method.

Yep, you were right on both counts. I meant the `new` keyword, and by asking "which method", I was looking for "constructor". But `__construct` makes sense too.

**Comment by paulstatezny** [comment]
Nice job on this! Take a look at my feedback and let me know if you understand it all, or if you have any questions. Once I know you have no more questions, I'll merge this.

**Comment by rdripley** [comment]
No that makes sense. I'm glad I was on the right track. I just did a pull request for section5.php so let me know when you have a chance to look at that.

**Pull request merged**